### DISCUSSION TITLE
Spot metering not working,

### DISCUSSION THREAD

**Discussion by DJane999** [gopro/labs#744] [Q&A]
I need the same spot metering point for exposure every time I use my cameras.

I load settings from the QR code, the settings take and I can see that they have changed.
but the spot metering setting for exposure does not take.
If I manually set a spot metering point, it is marked with brackets on screen. I load settings from QR code, the settings change, but the spot metering point does not change to the set point it stays on the old point.

If I don't manually set a spot metering point, then load settings from QR code,  how can i tell if the spot metering point has taken?  When I try this after a clean restart,  and load settings it is not marked by brackets, and when you go to set one it marks the new location where you have held your finger on the screen.

**Comment by dnewman-gpsw** [comment]
I'm not seeing this.  While the UI doesn't show the updated graphics overlays, Labs doesn't hack the UI, I see the spot meter coordinates for the exposure controls are updating with the QR Code.  

Should always list the camera model, and commands you are trying.  I just tested using HERO12.

**Reply by DJane999** [reply]
I'm using HERO12
I see the QR code change to,  
When I set a manual setting point using the camera screen it is marked persistently by brackets, showing the point.

**Reply by dnewman-gpsw** [reply]
Yes.  The QR Code is updating the coordinates used by the camera to set the exposure region, it just not updated the UI brackets.  Is that a problem?

**Reply by DJane999** [reply]
The problem would be not knowing if it is working correctly or not,
If it is saying one thing and doing another(or you hope it is, with no way of telling)

**Reply by dnewman-gpsw** [reply]
Spot meter via QR code is designed to be used in scripts, camera automation, where there is no reason to use the touchscreen. Most of Labs scripting doesn't provide UI feedback, as you are not expect to watch a camera running actions at particular times or triggers. When using the touchscreen there is very little reason to set a spot metering point via the QR Code. If you use both methods, it will be confusing, as Labs will not refresh the UI, so those brackets will not be updated. It is currently working as designed, as Labs works at a lower level than the UI.

**Reply by DJane999** [reply]
ok, got it, will trust that it is working

**Reply by nyetwurk** [reply]
Is there a way to tell if this persists after power off/on? Do you have to rescan it every time you turn on the gopro to restore spot metering?

**Reply by dnewman-gpsw** [reply]
You have to enabled it for every boot, however that can be done automatically with a boot script.

**Comment by TripleFJeff** [comment]
Is the spot metering setting recorded in the metadata? I have opened a file with the "metadata extraction" utility but it doesn't seem to be in there. 

I have a few settings QRs that I use to setup cameras that cover multiple items and I like to double check them, this is one of the few items that it seems there is no way to verify after the fact. I can see it when I scan the QR, just no way to check later on. I have had trouble in the past with settings not taking-- it seems to be solved (issue was that there was a limit to how long the strings could be, so I had to break it up into multiple QRs) but would like to be able to verify. I'm using a HERO8.

**Reply by dnewman-gpsw** [reply]
There is no metadata for spot metering.
